js/views/chatview.js

```diff
@@ -603,14 +611,15 @@
 			$el.find('.has-tooltip').tooltip({container: this._tooltipContainer});
 
 			var setAvatar = function($element, size) {
-				if ($element.data('user-id')) {
+				if ($element.data('user-id') && $element.data('user-id').substr(0, 6) !== 'guest/') {
 					if ($element.data('user-id') === 'all') {
 						$element.addClass('avatar icon icon-contacts');
 					} else {
 						$element.avatar($element.data('user-id'), size, undefined, false, undefined, $element.data('user-display-name'));
 					}
 				} else {
-					$element.imageplaceholder('?', $element.data('displayname'), size);
+					var firstChar = $element.data('user-display-name') ? $element.data('user-display-name').substr(0, 1) : 'G';
```

Comment:
Why the change from _?_ to _G_ for guests without a name set?

Reply:
We used G already for the guest at the chat input and in the participant list. So now the chat messages are consistent with that

Reply:
Hmm at least that is what Jan told me, but it seems it's not the case everywhere, let me check.

Reply:
No, the participant list and the avatar next to where the guest sets his name both use _?_; the chat messages, on the other hand, use _G_.

Reply:
Jan-C. Borchardt (16th July, 12:06)
> Also some stuff regarding guests:
> - Marco is shown with an M (in grey circle) in the right sidebar participant list, but with a "?" in grey circle in the chat. Should be an M in both cases, and also can be colored based on placeholder?
> - Also, would it be possible to @mention guests?

Reply:
So if a name is set we should use that letter, otherwise `G` or `?` @jancborchardt

Reply:
Reverted back to `?` as default, but it uses the first char, when a name is set.

Reply:
> Reverted back to `?` as default, but it uses the first char, when a name is set.

Yep, that’s perfect! Also then it should be grey when no name is set, but colored as soon as a name is set, is that possible?

Reply:
That is possible, but I'd like to quickly tell the difference between guests and normal users. Also having the guest avatar colored, would allow to impersonate users that have not uploaded an avatar, because your guest name will generate the same avatar.

Reply:
Maybe it’s good then if we always append ` (guest)` to the name then, or at least in the participant list like we do it with ` (moderator)`?

Cause most people in the world don’t have Nextcloud and will likely be guests – making that look nice too would be cool. :) (Also in the future it would also be cool to have avatar upload for guests etc. – ref Allow guest users to set avatar #357)